### ISSUE TITLE
Incorrect error handling with CLI argument parsing

### ISSUE THREAD

**Issue by rachitnigam** [fable-compiler/Fable#1180]
### Description

In compiled JS version of the program below, the try handler is not activated even when the program is not given any arguments.

### Repro code

```fsharp
[<EntryPoint>]
let main args =
    let (repetitions: float) = try float args.[0] with _ -> 1.0
    printf "%.9f\n" repetitions
    0
```

### Expected and actual results

Expected (from running with `fsharpc`): 1.0000
Actual: NaN

### Related information

* Fable version (`dotnet fable --version`): N/A. Using the `npm` distributed fable-compiler (v. 0.7.51) 
* Operating system: Ubuntu (17.04)

### Diagnostic information
The relevant part of the output from compiling the code is:
```javascript
  try {
    repetitions = Number.parseFloat(args[0]);
  } catch (matchValue) {
    repetitions = 20000000;
  }
```
Like everything in JavaScript, `Number.parseFloat` does not error when given an invalid input. The correct output should do something like:
```javascript
  try {
    repetitions = Number.parseFloat(args[0]);
    if (!repetitions) throw new Error(...)
  } catch (matchValue) {
    repetitions = 20000000;
  }
```

**Comment by alfonsogarciacaro** [comment]
I'd recommend using the new NETCore Fable, this issue should be fixed with latest version. It's been a while since we stopped updating the npm distribution (actually is already marked as deprecated). You can check how to get started with the new Fable here: http://fable.io/docs/getting-started.html

**Comment by rachitnigam** [comment]
Oh, okay. Thanks!
